common/lib/dependabot/dependency_file.rb

```diff
@@ -5,23 +5,29 @@
 module Dependabot
   class DependencyFile
     attr_accessor :name, :content, :directory, :type, :support_file,
-                  :symlink_target, :content_encoding, :deleted
+                  :symlink_target, :content_encoding, :operation
 
     class ContentEncoding
       UTF_8 = "utf-8"
       BASE64 = "base64"
     end
 
+    class Operation
+      UPDATE = "update"
+      CREATE = "create"
+      DELETE = "delete"
+    end
+
     def initialize(name:, content:, directory: "/", type: "file",
                    support_file: false, symlink_target: nil,
-                   content_encoding: ContentEncoding::UTF_8, deleted: false)
+                   content_encoding: ContentEncoding::UTF_8, operation: Operation::UPDATE)
```

Comment:
Unfortunately this initializer is used publicly, so this would result in a breaking change that I'm slightly uncomfortable with making right now. Adding a new param with a default value is fine, but we'll need to keep the `deleted:` param around to avoind breaking downstream dependants of dependabot-core.

We should start adding this to inline-docs to make that more clear (also for ourselves)

Reply:
Ok, I'll make sure deleted will stay and work

Reply:
@jurre Do you also want to keep the `attr_accessor`? Or just the param in the initializer?